app/code/Magento/CatalogGraphQl/etc/schema.graphqls

```diff
@@ -259,6 +259,7 @@ interface ProductInterface @typeResolver(class: "Magento\\CatalogGraphQl\\Model\
     meta_description: String @doc(description: "A brief overview of the product for search results listings, maximum 255 characters")
     image: String @doc(description: "The relative path to the main image on the product page")
     small_image: String @doc(description: "The relative path to the small image, which is used on catalog pages")
+    small_image_url: String @doc(description: "The small image URL, which is used on catalog pages") @resolver(class: "Magento\\CatalogGraphQl\\Model\\Resolver\\Product\\SmallImageUrl")
```

Comment:
Presence of both `small_image` and `small_image_url` may confuse developer. We should find another names or provide more details in the description. Also seems like similar pairs should be added for `image` and `thumbnail`. As an alternative solution, a path to media files may be part of a config type

Reply:
@rogyar how about having the following structure 
```
small_image: ProductImage
type ProductImage
{
   url: String
   path: String
}
```
 and then re-use it for other image types? 

@misha-kotov please provide a reference to the related tickets for other image types.

Reply:
Ticket for other image types https://github.com/magento/graphql-ce/issues/174